sdk/objectanchors/Azure.MixedReality.ObjectAnchors.Conversion/src/AssetConversionOperation.cs

```diff
@@ -0,0 +1,124 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Core.Pipeline;
+
+namespace Azure.MixedReality.ObjectAnchors.Conversion
+{
+    /// <summary>
+    /// A long running asset conversion process for Object Anchors
+    /// </summary>
+    public class AssetConversionOperation : Operation<AssetConversionProperties>
+    {
+        private static readonly TimeSpan defaultPollingInterval = TimeSpan.FromSeconds(15);
+        private readonly ObjectAnchorsConversionClient _objectAnchorsConversionClient;
+        private readonly Guid _jobId;
+        private Response<AssetConversionProperties> _lastConversionResponse;
+        private Response<AssetConversionProperties> _conclusiveConversionResponse;
+        private bool _conversionEnded;
+        private object _updateStatusLock = new object();
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AssetConversionOperation"/> class.
+        /// You must call <see cref="UpdateStatus(CancellationToken)"/> or <see cref="UpdateStatusAsync(CancellationToken)"/> before you can get the <see cref="Value"/>.
+        /// </summary>
+        /// <param name="jobId">The ID of this operation.</param>
+        /// <param name="objectAnchorsConversionClient">The client used to check for completion.</param>
+        public AssetConversionOperation(Guid jobId, ObjectAnchorsConversionClient objectAnchorsConversionClient)
+        {
+            _objectAnchorsConversionClient = objectAnchorsConversionClient;
+            _jobId = jobId;
+        }
+
+        /// <inheritdoc/>
+        public override string Id => _jobId.ToString();
+
+        /// <inheritdoc/>
+        public override AssetConversionProperties Value
+        {
+            get
+            {
+                lock (_lastConversionResponse)
```

Comment:
This will throw if `_lastConversionResponse` is null.

Reply:
Switched to the proper lock (this was a bug)

Reply:
Fixed to correct lock object